### PR TITLE
let users open urls from dapps view

### LINKED ISSUE #3957: URL input on Applications tab
...to go directly to URL from `Applications` tab.

Requires including browsing address in routing, i.e. `#/web/https://mkr.market`.

screenshot from 2016-12-23 12 59 14

After: #3956

### PR DESCRIPTION
**Pull request by derhuerst**
Fixes #3957.

### CODE REVIEW

File: js/src/views/Dapps/UrlButton/urlButton.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import { withRouter } from 'react-router';
+
+import Button from '~/ui/Button';
+import { LinkIcon } from '~/ui/Icons';
+import Input from '~/ui/Form/Input';
+
+import styles from './urlButton.css';
+
+class UrlButton extends Component {
+  static propTypes = {
```
Comment (jacogr) [inline js/src/views/Dapps/UrlButton/urlButton.js]: router is actually also available on the context, so we don't need the withRouter wrapper. Could just do `static contextTypes = { router: PropTypes.object.isRequired }`, e.g. Same as what we do in `views/Settings/settings.js`

File: js/src/views/Dapps/UrlButton/urlButton.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      <Input
+        hint={
+          <FormattedMessage
+            id='dapps.button.url.input'
+            defaultMessage='https://mkr.market' />
+        }
+        onBlur={ this.hideInput }
+        onFocus={ this.showInput }
+        onSubmit={ this.inputOnSubmit }
+        style={ { display: 'inline-block', width: '20em' } }
```
Comment (jacogr) [inline js/src/views/Dapps/UrlButton/urlButton.js]: Small picky thing, but wouldn't it make more sense to define the style as a constant and then just reference it here?

File: js/src/views/Dapps/dapps.js [hunk 3]
```diff
@@ -88,6 +89,9 @@ class Dapps extends Component {
               defaultMessage='Decentralized Applications' />
           }
           buttons={ [
+            <UrlButton
```
Comment (jacogr) [inline js/src/views/Dapps/dapps.js]: Think this could just be a single line since there is only one attribute

File: js/src/views/Web/web.js [hunk 4]
```diff
@@ -98,18 +103,23 @@ export default class Web extends Component {
...
 
     this.setState({
       isLoading: true,
-      url: `${displayedUrl}${separator}t=${Date.now()}`
+      url: formatUrl(parsed)
     });
   };
 
-  handleIframeLoad = (ev) => {
+  iframeOnLoad = () => {
```
Comment (jacogr) [inline js/src/views/Web/web.js]: Possibly `onIframeLoad` since it is an event?

**Comment by jacogr** [comment]
Small grumbles, but looking quite good. (Just want to play a bit to see if anything pops out)

File: js/src/views/Dapps/UrlButton/urlButton.js (hunk 1 above)
Comment (derhuerst) [inline js/src/views/Dapps/UrlButton/urlButton.js]: Followed [the a `react-router` guide](https://github.com/ReactTraining/react-router/blob/c3cd9675bd8a31368f87da74ac588981cbd6eae7/upgrade-guides/v2.4.0.md#withrouter-hoc-higher-order-component), which says this:

> Prior to 2.4.0, you could access the `router` object via [`this.context`](https://facebook.github.io/react/docs/context.html). This is still true, but `context` is often times a difficult and error-prone API to work with.

> In order to more easily access the `router` object, a `withRouter` higher-order component has been added as the new primary means of access.

> **It's important to note this is not a deprecation of the `context` API.** As long as React supports `this.context` in its current form, any code written for that API will continue to work. We will continue to use it internally and you can continue to write in that format, if you want. We think this new HoC is nicer and easier, and will be using it in documentation and examples, but it is not a hard requirement to switch.

File: js/src/views/Web/web.js (hunk 4 above)
Comment (derhuerst) [inline js/src/views/Web/web.js]: But it is a `load` even *on* the `iframe`, not a `iframeLoad` event. I think the current form is more expressive.

File: js/src/views/Web/web.js [hunk 5]
```diff
@@ -67,28 +64,44 @@ export default class Web extends Component {
...
+    const { url } = this.state;
+    if (!url || !token) {
+      return null;
+    }
+
+    const parsed = parseUrl(url);
+    let host = parsed.host;
+    let path = parsed.path;
+    if (!host) {
+      host = parsed.path.split(pathSep).slice(0, 1);
```
Comment (ngotchac) [inline js/src/views/Web/web.js]: Isn't this returning an array ? It's fine with the cast to String though, but still...

File: js/src/views/Web/web.js [hunk 6]
```diff
@@ -67,28 +64,44 @@ export default class Web extends Component {
...
+      path = parsed.path.split(pathSep).slice(1).join(pathSep);
```
Comment (ngotchac) [inline js/src/views/Web/web.js]: Isn't there missing the first `/` then ? Try `reddit.com/r/ethereum`

Also, what's the point of using `pathSep` ? Should always be `/` with URLs, no ?

Comment (derhuerst) [inline js/src/views/Web/web.js]: > Also, what's the point of using pathSep ? Should always be / with URLs, no ?

You're right. I am so much into "ways use `path.sep` to make modules cross-platform" that i didn't think about it.

**Pull request merged**